readthedocs/organizations/tests/test_privacy_urls.py

```diff
@@ -74,3 +76,195 @@ def test_public_urls(self):
+    }
+
+    def login(self):
+        self.client.force_login(self.another_user)
+
+    def test_public_urls(self):
+        from readthedocs.organizations.urls.public import urlpatterns
+
+        self._test_url(urlpatterns)
+
+
+@override_settings(
+    RTD_ALLOW_ORGANIZATIONS=True,
+    ALLOW_PRIVATE_REPOS=True,
+)
+class AnotherUserWithPrivateOrganizationsTest(OrganizationMixin, TestCase):
+
+    """If organizations are private, an user outside the organization can't access the public views."""
+
+    default_status_code = 404
+    response_data = {
+        # Places where we 302 on success.
+        "/organizations/invite/{hash}/redeem/": {"status_code": 302},
+    }
+
+    def login(self):
+        self.client.force_login(self.another_user)
+
+    def test_public_urls(self):
+        from readthedocs.organizations.urls.public import urlpatterns
+
+        self._test_url(urlpatterns)
+
+
+@override_settings(
+    RTD_ALLOW_ORGANIZATIONS=True,
+    ALLOW_PRIVATE_REPOS=False,
+)
+class AnotherUserWithPublicOrganizationsPrivateViewsTest(OrganizationMixin, TestCase):
+
+    """If organizations are public, an user outside the organization can access the public views."""
+
+    default_status_code = 404
+    response_data = {
+        # All users have access to these views.
+        "/organizations/": {"status_code": 200},
+        "/organizations/create/": {"status_code": 200},
+        "/organizations/verify-email/": {"status_code": 200},
+        # 405's where we should be POST'ing
+        "/organizations/{slug}/owners/{owner}/delete/": {"status_code": 405},
+        "/organizations/{slug}/teams/{team}/members/{member}/revoke/": {
+            "status_code": 405
+        },
+    }
+
+    def login(self):
+        self.client.force_login(self.another_user)
+
+    def test_private_urls(self):
+        from readthedocs.organizations.urls.private import urlpatterns
+
+        self._test_url(urlpatterns)
+
+
+@override_settings(
+    RTD_ALLOW_ORGANIZATIONS=True,
+    ALLOW_PRIVATE_REPOS=True,
+)
+class AnotherUserWithPrivateOrganizationsPrivateViewsTest(OrganizationMixin, TestCase):
+
+    """If organizations are private, an user outside the organization can't access the private views."""
+
+    default_status_code = 404
+    response_data = {
+        # All users have access to these views.
+        "/organizations/": {"status_code": 200},
+        "/organizations/create/": {"status_code": 200},
+        "/organizations/verify-email/": {"status_code": 200},
+        # 405's where we should be POST'ing
+        "/organizations/{slug}/owners/{owner}/delete/": {"status_code": 405},
+        "/organizations/{slug}/teams/{team}/members/{member}/revoke/": {
+            "status_code": 405
+        },
+    }
+
+    def login(self):
+        self.client.force_login(self.another_user)
+
+    def test_private_urls(self):
+        from readthedocs.organizations.urls.private import urlpatterns
+
+        self._test_url(urlpatterns)
```

Comment:
Pretty hard to follow all these tests with such magic. None of them seems to do anything different one from the other (just a setting and URLs). I assume that _all this magic_ is inside this method :/

I assume this is fine, tho, to avoid repetition, but... :(